src/wayland/drm_syncobj/mod.rs

```diff
@@ -0,0 +1,340 @@
+//! DRM syncobj protocol
+
+use std::{cell::RefCell, os::unix::io::AsFd};
+use wayland_protocols::wp::linux_drm_syncobj::v1::server::{
+    wp_linux_drm_syncobj_manager_v1::{self, WpLinuxDrmSyncobjManagerV1},
+    wp_linux_drm_syncobj_surface_v1::{self, WpLinuxDrmSyncobjSurfaceV1},
+    wp_linux_drm_syncobj_timeline_v1::{self, WpLinuxDrmSyncobjTimelineV1},
+};
+use wayland_server::{
+    protocol::wl_surface::WlSurface, Client, DataInit, Dispatch, DisplayHandle, GlobalDispatch, New,
+    Resource, Weak,
+};
+
+use super::{
+    compositor::{self, with_states, BufferAssignment, Cacheable, SurfaceAttributes},
+    dmabuf::get_dmabuf,
+};
+use crate::backend::drm::DrmDeviceFd;
+
+mod sync_point;
+pub use sync_point::{DrmSyncPoint, DrmTimeline};
+
+/// Handler trait for DRM syncobj protocol.
+pub trait DrmSyncobjHandler {
+    // TODO better way to deal with this?
+    /// DRM device for importing syncobj file descriptors
+    fn import_device(&self) -> &DrmDeviceFd;
```

Comment:
Can't we at least pass the `Client` here? I am not sure, if any part of the stack expects a particular device, but I would like to make this the same device we advertise via wl_drm or as a main device in cosmic's implementation.

Reply:
Good point. I *think* it shouldn't matter? In which case it would be simpler to take a `DrmDeviceFd` when creating the global.

But if it does matter, that a reason to stick with a trait method like this. Although, we probably would want to derive the node from the `Dmabuf`. Which should handle that, but also a device specified with https://github.com/Smithay/smithay/pull/1430. Right?

We could pass the `WlBuffer`. But if the implementation here assumes the buffer is created with `zwp_linux_dmabuf_v1`, we could pass the `Option<DrmNode>`.

Reply:
> [...] we could pass the `Option<DrmNode>`

That would only be set for dmabuf <= 3 (and potentially >= 6), right? We have more information given the main device in cosmic. So imo the `WlBuffer` is the most flexible, as it also allows to get the `Client`.

Whether we make use of this in the end is a separate question, but I think we should offer this in the api.